lib/Transforms/LoopFusion.cpp

```diff
@@ -1459,8 +1470,10 @@ struct GreedyFusion {
   // *) Third pass fuses any remaining producer nodes into their users.
   void run() {
     // TODO(andydavis) Run this repeatedly until a fixed-point is reached.
-    fuseProducerConsumerNodes(/*maxSrcUserCount=*/1);
-    fuseSiblingNodes();
+    if (clSiblingFusion) {
+      fuseProducerConsumerNodes(/*maxSrcUserCount=*/1);
```

Comment:
I assumed that if sibling fusion was disabled, only one `fuseProducerConsumerNodes` pass was needed and I left the one with the highest `maxSrcUserCount`. I now see comment in line 1472 and I'm not sure if a single pass will address all the producer-consumer cases. Could a first pass expose new producer-consumer cases that are not handled by the own pass?

Reply:
Yes, please put call the "fuseSiblingNodes()" under the flag "clSiblingFusion", otherwise I think users will get confused. If you need to disable producer-consumer fusion, then it should be under a separate flag.

Reply:
However, we may not need the flag at all, if you can construct a test case which is illegal for sibling fusion.

// producer loop
for ...
  store %v, %a[%i]

// Consumer loop 1
for ...
  load %a[%i]
  store %v, %b[%i]

// Consumer loop 2
for ...
  load %a[%i]
  load %b[%i]

The dependence on '%b' between the two consumer nodes should prevent sibling fusion